zipline/data/hdf5_daily_bars.py

```diff
@@ -0,0 +1,668 @@
+from functools import partial
+
+import h5py
+import logbook
+import numpy as np
+import pandas as pd
+from six import iteritems
+from six.moves import reduce
+
+from zipline.data.bar_reader import NoDataBeforeDate, NoDataAfterDate
+from zipline.data.session_bars import SessionBarReader
+from zipline.utils.memoize import lazyval
+
+
+log = logbook.Logger('HDF5DailyBars')
+
+
+DATA = 'data'
+INDEX = 'index'
+LIFETIMES = 'lifetimes'
+
+SCALING_FACTOR = 'scaling_factor'
+
+OPEN = 'open'
+HIGH = 'high'
+LOW = 'low'
+CLOSE = 'close'
+VOLUME = 'volume'
+
+FIELDS = (OPEN, HIGH, LOW, CLOSE, VOLUME)
+
+DAY = 'day'
+SID = 'sid'
+
+START_DATE = 'start_date'
+END_DATE = 'end_date'
+
+
+scaling_factors = {
```

Comment:
shouldn't we read this out of the file's metadata?

Reply:
These are the factors that get written to the file (this is hardcoded right now but I plan to make it a parameter with these as the defaults). In the reader, they are read out of the file [here](https://github.com/quantopian/zipline/pull/2295/files/d754fb93b29f6069be49fff1a6f99b6f63aa34ff#diff-1a70086405577f2f9433c7624909ec30R261).